src/twisted/names/secondary.py

```diff
@@ -128,20 +149,26 @@ def fromServerAddressAndDomain(cls, serverAddress, domain):
             transfers will be attempted from.
 
         @param domain: A C{bytes} giving the domain to transfer.
+        @type domain: L{bytes}
 
         @return: A new instance of L{SecondaryAuthority}.
         """
-        secondary = cls(None, None)
-        secondary.primary = serverAddress[0]
-        secondary._port = serverAddress[1]
-        secondary.domain = domain
+        primary, port = serverAddress
+        secondary = cls(primary, domain)
+        secondary._port = port
         return secondary
 
 
     def transfer(self):
-        if self.transferring:
+        """
+        Attempt a zone transfer.
+
+        @returns: A L{Deferred} that fires with L{None} when attempted zone
+            transfer has completed.
+        """
+        if self.transferring:  # <-- never true
             return
```

Comment:
This seems like it would make the docstring a lie, if it actually worked 😬

Reply:
I'm afraid that :grimacing: summarizes this module too well. :(